src/functions-reference/higher-order_functions.Rmd

```diff
@@ -126,14 +144,23 @@ solver directly to the system function.
 
 #### Algorithmic details
 
-The algebraic solver uses the Powell hybrid method [@Powell:1970],
+Stan offers two algebraic solvers.
+The first one, `algebra_solver`, uses the Powell hybritd method [@Powell:1970],
 which in turn uses first-order derivatives. The Stan code builds on
 the implementation of the hybrid solver in the unsupported module for
 nonlinear optimization problems of the Eigen library [@Eigen:2013].
 This solver is in turn based on the algorithm developed for the
 package MINPACK-1 [@minpack:1980].
 
-The Jacobian of the solution with respect to auxiliary parameters is
+The second solver, `algebra_solver_newton`, uses Newton's method,
+also a first-order derivative based numerical solver.
+The Stan code builds on the implementation in KINSOL
+from the SUNDIALS suite [@Hindmarsh:2005].
+In general, this method works better than Powell's algorithm,
```

Comment:
Ui, is it that much better in general? In what terms is it better - faster, more precise or robust or even all of that?

Reply:
I'm not 100% sure. In the settings I tested, the Newton solver has always been faster (2 - 20 times), and returned a answer consistent with Powell's method. The examples I looked at were PK steady states and Laplace approximations. The dogleg solvers may be better when the initial guess is far from the solution.

It's possible that KINSOL is better implemented than Eigen, so the difference is an engineering one, as opposed to simply an algorithmic one.

@yizhang-yiz probably has a better understanding of the matter than I do, and I can update the documentation accordingly.